### DISCUSSION TITLE
groupId for KafkaUsers

### DISCUSSION THREAD

**Discussion by mumlawski** [strimzi/strimzi-kafka-operator#10340] [Q&A] [answered]
Hi all,

My question is related to groupIds (https://kafka.apache.org/documentation/#consumerconfigs_group.id)

I went through Strimzi docs and from my understand now, Strimzi operator doesn't manage consumer groups directly. I notice it creates consumer or producer groups automatically with some random Ids added to identify them.

Is there any plan in the future to allow to create/manage user groups directly from Strimzi? i.e. in the KafkaUser spec or other?

Thanks!

**Comment by scholzj** [comment, marked as answer]
Strimzi does not do anything with consumer groups. Consumer groups in Kafka are not user groups. They are used to coordinate various groups of consumers. What is the consumer group to which a consumer belongs is not defined on the user level but in the consumer configuration independently on the user's identity. So if you see _random IDs_ as the consumer groups, that is a question of the application configuration.

The only place where consumer groups relate to users is that if you have enabled authorization, the user needs to be granted the right to use a given consumer group. If you use the `type: simple` authorization, this right can be granted in the `KafkaUser` resource. If you use another authorization type, it has to be granted there.

**Reply by mumlawski** [reply]
I think this makes sense, thanks for explanation.
